src/containers/load-record.js

```diff
@@ -5,105 +5,23 @@
  */
 
 import React from 'react'
-import { request, isValidSlug } from './request'
-import { type Record } from '../record'
+import DataFetcher from './data-fetcher'
+import OptionsContext from '../contexts/options'
 
-type Props = {
-  makeURL: (string, ?mixed) => string,
-  onError: (*) => Error,
-  onFetch: (*) => Record,
-  url: string,
-  slug: ?mixed,
-  children: ?(Result) => *
+function isValidSlug(slug: any): boolean {
```

Comment:
Should live in the DataFetcher class? I see this defined twice, and used in classes that extend DataFetcher.

Reply:
This should only be necessary for the LoadRecord component. Where else do you see this defined?

Reply:
Nevermind! I'm not sure what I saw now. Maybe just deleted code :)

Reply:
No worries. It's a bigger diff than I wanted. Ideally I should have split out the Flow update and the context change into separate PRs 🙈